code-studio/src/js/progress.js

```diff
@@ -153,6 +154,31 @@ progress.renderCourseProgress = function (scriptData, currentLevelId) {
   );
 };
 
+// Return the level with the highest progress, or the first level if none have
+// been attempted
+progress.bestResultLevelId = function (levelIds, progress) {
+  // The usual case
+  if (levelIds.length === 1) {
+    return levelIds[0];
+  }
+
+  // Return the level with the highest result
+  var attemptedIds = levelIds.filter(function (id) { progress[id] !== 0; });
```

Comment:
this either needs a return OR (preferred) to just be an arrow function: `levelIds.filter(id => progress[id] !== 0);`